### PR TITLE
Helm Chart: Add Routes and set enable PersistentStorage by default

### PR DESCRIPTION
**Pull request by mphammer**
# Pull Request template

**Link to github issue (if applicable):**

* N/A
* JIRA: https://jira-sig.internal.synopsys.com/browse/IALERT-1681

**If nothing above, what is your reason for this pull request:**

1. Customers should be able to deploy with Routes if they are in openshift
2. Customers should always be deploying their resources with persistent storage. This is also how the Black Duck Helm Chart is implemented. 

**Changes proposed in this pull request:**

* Add the Route manifest for the Route Resource
* Add logic to enable/disable the route, or use a service for k8s
* Change the default "enablePersistentStorage" to be true

### CODE REVIEW

File: deployment/helm/templates/alert.yaml [hunk 1]
```diff
@@ -131,31 +155,26 @@ spec:
   type: ClusterIP
 ---
 
-{{- if and .Values.enablePersistentStorage (not .Values.persistentVolumeClaimName) }}
-apiVersion: v1
-kind: PersistentVolumeClaim
+{{ if and .Values.exposeui (eq .Values.exposedServiceType "Openshift" ) -}}
```
Comment (msenmurugan) [inline deployment/helm/templates/alert.yaml:158]: can you pls change from Openshift to OpenShift?

File: deployment/helm/templates/alert.yaml [hunk 2]
```diff
@@ -131,31 +155,26 @@ spec:
...
+apiVersion: route.openshift.io/v1
+kind: Route
 metadata:
   labels:
     app: alert
+    component: route
     name: {{ .Release.Name }}
-  name: {{ .Release.Name }}-alert-pvc
+  name: {{ .Release.Name }}-alert-exposed
```
Comment (msenmurugan) [inline deployment/helm/templates/alert.yaml:166]: can you pls remove "-exposed" from the name of the route

File: deployment/helm/templates/alert.yaml [hunk 3]
```diff
@@ -131,31 +155,26 @@ spec:
...
-{{- end }}
-
-{{ if .Values.exposeui -}}
+  port:
+    targetPort: port-{{ .Values.alert.port }}
+  tls:
+    termination: passthrough
+  to:
+    kind: Service
+    name: {{ .Release.Name }}-alert
```
Comment (msenmurugan) [inline deployment/helm/templates/alert.yaml:175]: can you please check whether an Alert service name is correct?

File: deployment/helm/values.yaml [hunk 4]
```diff
@@ -50,7 +50,7 @@ status: Running
 
 # Expose Alert's User Interface
 exposeui: true
-exposedServiceType: NodePort # ClusterIP | NodePort | LoadBalancer
+exposedServiceType: NodePort # ClusterIP | NodePort | LoadBalancer | Openshift
```
Comment (msenmurugan) [inline deployment/helm/values.yaml:53]: remove ClusterIP from it and also change Openshift to OpenShift

File: deployment/helm/templates/alert.yaml (hunk 3 above)
Comment (mphammer) [reply deployment/helm/templates/alert.yaml:175]: It matches the name of the ClusterIP service that is always created with Alert

File: deployment/helm/values.yaml (hunk 4 above)
Comment (mphammer) [reply deployment/helm/values.yaml]: Done!

File: deployment/helm/templates/alert.yaml (hunk 1 above)
Comment (mphammer) [reply deployment/helm/templates/alert.yaml]: Done!

File: deployment/helm/templates/alert.yaml (hunk 3 above)
Comment (mphammer) [reply deployment/helm/templates/alert.yaml:175]:
```
apiVersion: v1
kind: Service
metadata:
  labels:
    app: alert
    name: {{ .Release.Name }}
  name: {{ .Release.Name }}-alert
  namespace: {{ .Release.Namespace }}
```

**Comment by mphammer** [comment]
This PR will be implemented in https://github.com/blackducksoftware/blackduck-alert/pull/941

**Pull request closed without merging**